app/controllers/GroupSchemeController.scala

```diff
@@ -121,142 +93,119 @@ trait GroupSchemeController extends ERSReturnBaseController with Authenticator w
+          requestObject.getSchemeId match {
+            case PageBuilder.SCHEME_CSOP =>
+              formData.groupScheme match {
+                case Some(PageBuilder.OPTION_YES) => Redirect(routes.GroupSchemeController.manualCompanyDetailsPage())
+                case _ => Redirect(routes.AltAmendsController.altActivityPage())
               }
-            }
-            case PageBuilder.SCHEME_SAYE => {
-              formData.groupScheme.get match {
-                case PageBuilder.OPTION_NO => Redirect(routes.AltAmendsController.altActivityPage())
-                case PageBuilder.OPTION_YES => Redirect(routes.GroupSchemeController.manualCompanyDetailsPage())
+            case PageBuilder.SCHEME_SAYE =>
+              formData.groupScheme match {
+                case Some(PageBuilder.OPTION_YES) => Redirect(routes.GroupSchemeController.manualCompanyDetailsPage())
+                case _ => Redirect(routes.AltAmendsController.altActivityPage())
               }
-            }
-            case PageBuilder.SCHEME_EMI => {
-              formData.groupScheme.get match {
-                case PageBuilder.OPTION_NO => Redirect(routes.SummaryDeclarationController.summaryDeclarationPage())
-                case PageBuilder.OPTION_YES => Redirect(routes.GroupSchemeController.manualCompanyDetailsPage())
+            case PageBuilder.SCHEME_EMI =>
+              formData.groupScheme match {
+                case Some(PageBuilder.OPTION_YES) => Redirect(routes.GroupSchemeController.manualCompanyDetailsPage())
+                case _ => Redirect(routes.SummaryDeclarationController.summaryDeclarationPage())
               }
-            }
-            case PageBuilder.SCHEME_SIP => {
-              formData.groupScheme.get match {
-                case PageBuilder.OPTION_NO => Redirect(routes.TrusteeController.trusteeDetailsPage())
-                case PageBuilder.OPTION_YES => Redirect(routes.GroupSchemeController.manualCompanyDetailsPage())
+            case PageBuilder.SCHEME_SIP =>
+              formData.groupScheme match {
+                case Some(PageBuilder.OPTION_YES) => Redirect(routes.GroupSchemeController.manualCompanyDetailsPage())
+                case _ => Redirect(routes.TrusteeController.trusteeDetailsPage())
               }
-            }
-            case PageBuilder.SCHEME_OTHER => {
-              formData.groupScheme.get match {
-                case PageBuilder.OPTION_NO => Redirect(routes.SummaryDeclarationController.summaryDeclarationPage())
-                case PageBuilder.OPTION_YES => Redirect(routes.GroupSchemeController.manualCompanyDetailsPage())
+            case PageBuilder.SCHEME_OTHER =>
+              formData.groupScheme match {
+                case Some(PageBuilder.OPTION_YES) => Redirect(routes.GroupSchemeController.manualCompanyDetailsPage())
+                case _ => Redirect(routes.SummaryDeclarationController.summaryDeclarationPage())
```

Comment:
This could all collapse into something quite easy to read

`(requestObject.getSchemeId, formData.groupScheme) match {

case (_, Some(PageBuilder.OPTION_YES) => Redirect(routes.GroupSchemeController.manualCompanyDetailsPage())
case (CSOP | SAYE, _) => AltAmendsController
case (EMI | SCHEME_OTHER, _) => SummaryDeclaration
case (SCHEME_SIP, _) => TrusteeController
case (_, _) //if the compiler throws a warning.
...
case (_, _) => 

}`